include/nbl/asset/ICPUBuffer.h

```diff
@@ -107,7 +99,37 @@ class ICPUBuffer : public asset::IBuffer, public asset::IAsset
             return true;
         }
 
+        bool equals(const IAsset* _other) const override
+		{
+			auto* other = static_cast<const ICPUBuffer*>(_other);
+			return compatible(other) && (m_creationParams.usage.value == other->m_creationParams.usage.value);
+		}
+
+
+		bool canBeRestoredFrom(const IAsset* _other) const override
+		{
+			auto* other = static_cast<const ICPUBuffer*>(_other);
+			return compatible(other);
+		}

```

Comment:
can we just get rid of this, and replace its usages with `compatible` ?

Reply:
hmm probably not the case for other higher level assets

Reply:
https://discord.com/channels/593902898015109131/1078226704092381194/1083069244650045600